### PR TITLE
Support log scale in parallel coordinate

### PR DESCRIPTION
**Pull request by tohmae**
This PR addresses issue #669 .

1. optuna.visualization.plot_parallel_coordinate
- before
before

- after
after

2. optuna.visualization.matplotlib.plot_parallel_coordinate
- before
matplotlib_before

- after
matplotlib_after

### CODE REVIEW

**Comment by hvy** [comment]
Thanks @tohmae for tackling the long-standing issue. @HideakiImamura could you have a look at this change?

File: tests/visualization_tests/test_parallel_coordinate.py [hunk 1]
```diff
@@ -121,3 +124,51 @@ def fail_objective(_: Trial) -> float:
...
+    assert figure.data[0]["dimensions"][1]["label"] == "param_a"
+    assert figure.data[0]["dimensions"][1]["range"] == (-6.0, -4.0)
+    assert figure.data[0]["dimensions"][1]["values"] == (-6, math.log10(2e-5), -4)
+    assert figure.data[0]["dimensions"][1]["ticktext"] == ("1e-06", "1e-05", "0.0001")
+    assert figure.data[0]["dimensions"][1]["tickvals"] == (-6, -5, -4.0)
+    assert figure.data[0]["dimensions"][2]["label"] == "param_b"
+    assert figure.data[0]["dimensions"][2]["range"] == (1.0, math.log10(200))
+    assert figure.data[0]["dimensions"][2]["values"] == (1.0, math.log10(200), math.log10(30))
+    assert figure.data[0]["dimensions"][2]["ticktext"] == ("10", "100", "200")
+    assert figure.data[0]["dimensions"][2]["tickvals"] == (1.0, 2.0, math.log10(200))
```
Comment (HideakiImamura) [inline tests/visualization_tests/test_parallel_coordinate.py:174]: Ditto.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR! I have some early comments. Please take a look.

**Comment by HideakiImamura** [comment]
@tohmae The CI is failing, but this is not due to this PR change, so don't worry about it.

**Comment by HideakiImamura** [comment]
The failure of the CI is fixed in #2166. Could you merge the master branch?

**Review by HideakiImamura — APPROVED** [review]
Thanks for the quick fix! LGTM!

**Comment by HideakiImamura** [comment]
@himkt Please merge this PR after your review.

**Review by himkt — APPROVED** [review]
Thank you for your contribution.
Let me rebase the latest master for fixing the conflict.

**Pull request merged**